src/SWP/Bundle/WebRendererBundle/Theme/Configuration/TenantableConfigurationProvider.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Web Renderer Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\WebRendererBundle\Theme\Configuration;
+
+use SWP\Bundle\WebRendererBundle\Theme\Helper\ThemeHelperInterface;
+use Sylius\Bundle\ThemeBundle\Configuration\ConfigurationProviderInterface;
+use Sylius\Bundle\ThemeBundle\Configuration\Filesystem\ConfigurationLoaderInterface;
+use Sylius\Bundle\ThemeBundle\Locator\FileLocatorInterface;
+
+class TenantableConfigurationProvider implements ConfigurationProviderInterface
+{
+    /**
+     * @var FileLocatorInterface
+     */
+    private $fileLocator;
+
+    /**
+     * @var ConfigurationLoaderInterface
+     */
+    private $loader;
+
+    /**
+     * @var string
+     */
+    private $configurationFilename;
+
+    /**
+     * @var ThemeHelperInterface
+     */
+    private $themeHelper;
+
+    /**
+     * @param FileLocatorInterface         $fileLocator
+     * @param ConfigurationLoaderInterface $loader
+     * @param string                       $configurationFilename
+     * @param ThemeHelperInterface         $themeHelper
+     */
+    public function __construct(
+        FileLocatorInterface $fileLocator,
+        ConfigurationLoaderInterface $loader,
+        $configurationFilename,
+        ThemeHelperInterface $themeHelper
+    ) {
+        $this->fileLocator = $fileLocator;
+        $this->loader = $loader;
+        $this->configurationFilename = $configurationFilename;
+        $this->themeHelper = $themeHelper;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigurations()
+    {
+        try {
+            $configs = array_map(
+                [$this->loader, 'load'],
+                $this->fileLocator->locateFilesNamed($this->configurationFilename)
+            );
+
+            return array_map(
+                [$this->themeHelper, 'process'],
+                $configs
+            );
+        } catch (\InvalidArgumentException $exception) {
```

Comment:
good catch, its because I wanted to silence it and moved the logic to custom theme context, but I think here its fine that it throws it.